sdk/src/simple_vote_transaction_checker.rs

```diff
@@ -45,7 +45,7 @@ pub fn is_simple_vote_transaction_impl<'a>(
         && is_legacy_message
         && instruction_programs
             .next()
-            .xor(instruction_programs.next())
             .map(|program_id| program_id == &solana_sdk::vote::program::ID)
             .unwrap_or(false)
+        && instruction_programs.next().is_none()
```

Comment:
would rather leave this as is, @tao-stones thoughts?

My reasoning is this.
As is, with xor, we will check the number of instructions before doing the pubkey comparison. Pubkey comparison is (probably?) more expensive than checking a simple is next item (most cases simple check on integer), so it's preferable to do that only if we're sure we need to.

Reply:
I meant the same with my comment above - `xor` before pubkey comparing is a performance optimization. Given this function is on hotpath, agree w @apfitzge to leave as is.